geode-core/src/main/java/org/apache/geode/internal/admin/SSLConfig.java

```diff
@@ -237,10 +249,26 @@ public void toDSProperties(Properties props) {
     public Builder() {}
 
     public SSLConfig build() {
+
+      final String sniProxyProperty = System.getProperty(GEMFIRE_PREFIX + "security.sni-proxy");
+      final InetSocketAddress sniProxyAddress;
+      if (sniProxyProperty == null) {
+        sniProxyAddress = null;
+      } else {
+        try {
+          final URL parser = new URL("http://" + sniProxyProperty);
```

Comment:
Good point. There are many host+port parsers in the codebase. The one I was thinking of was the one that parses locator host+port: `GMSUtil.parseLocators()`. I opted not to use that one because it dictates a nonstandard syntax e.g. `hostname[123]`. That format is used in a few places in Geode but it's unusual to folks who are used to web browsers.

Do you know, offhand, of another parser we've got that parses the more orthodox colon-separated port (and supports fqdn's, IPv4 addys and IPv6 ones too?)

Reply:
I found many host+port parsers but none jumped out as the one to reuse. I'd like to go ahead with the use of the `URL` class for this PR and come in with a more fulsome approach in the follow-on story: [GEODE-7852](https://issues.apache.org/jira/browse/GEODE-7852). That story will replace the system property with a connection pool configuration property.